libraries/chef_f5.rb

```diff
@@ -41,6 +44,33 @@ def pool_is_missing_node?(pool, node)
     members.map { |m| m[:address] }.grep(/#{with_partition node}/).empty?
   end
 
+  def pool_is_missing_monitor?(pool, monitor)
+    monitors = api.LocalLB.Pool.get_monitor_association(pool_names:
+      {item: with_partition(pool)}
+    )[:item]
+
+    monitors = [ monitors ] if monitors.is_a? Hash
+    monitors.select do |mon|
+      mon[:monitor_rule][:monitor_templates][:item] == with_partition(monitor)
+    end.empty?
+  end
+
+  def add_monitor(pool, monitor)
+    api.LocalLB.Pool.set_monitor_association(monitor_associations: {
+      item: [
+        {pool_name: pool,
+         monitor_rule: {
+           monitor_templates: {
+             item: monitor
+           }
+         },
+         quorum: '0',
+         type: 'MONITOR_RULE_TYPE_SINGLE'
```

Comment:
What `set_monitor_association` does is remove all monitors, and creates a single association to the given monitor. If the parameter `monitor` is an array of monitors, then a monitor association is created with an array of monitor templates, and the type is automatically set to `MONITOR_RULE_TYPE_AND_LIST`, regardless of the type passed in. A pool can have only 1 monitor association. Therefore, if `monitor` is an array and for each item in `monitor` you created a new monitor association only the last one is used. I confirmed all this by actually testing it. To sum it up: **this method sets the monitor, not adds a monitor**. To _add_ a monitor you would just have to query iControl to get the current monitors and append the new monitor to the current monitors in the call to `set_monitor_association`